comm/src/main/scala/coop/rchain/comm/transport/SslSessionServerInterceptor.scala

```diff
@@ -27,42 +28,55 @@ class SslSessionServerInterceptor() extends ServerInterceptor {
       call: ServerCall[ReqT, RespT]
   ) extends ServerCall.Listener[ReqT] {
 
-    override def onHalfClose(): Unit = next.onHalfClose()
-    override def onCancel(): Unit    = next.onCancel()
-    override def onComplete(): Unit  = next.onComplete()
-    override def onReady(): Unit     = next.onReady()
+    @volatile
+    private var closeWithStatus = Option.empty[Status]
```

Comment:
What's the idea behind this?

Reply:
@rabbitonweb A `ServerCall` is kind of a state machine. It goes thru the states (omitting `Cancel`):
- `Ready`: connection established, ready to receive messages
- `Message`: receive one or more messages
- `HalfClose`: client signals that all messages have been sent and he's waiting for a response
- `Complete`: the response has been sent to the client

We inspect all incoming messages in the interceptor hook `OnMessage`. However, we can't send a response (in our case a failure status) to the client right away. We need to wait for the `OnHalfClose` hook to send the response. That's why we need a `var`. TBH, I think the `@volatile` annotation is not required but it does no harm. Please keep in mind that an `InterceptionListener` instance is created for every single `ServerCall` so the `var` is not shared between calls.